pennylane/tape/tape.py

```diff
@@ -237,6 +237,47 @@ def stop_at(obj):  # pylint: disable=unused-argument
     return new_tape
 
 
+def expand_tape_state_prep(tape, skip_first=True):
+    """Expand all instances of StatePrep operations in the tape.
+
+    Args:
+        tape (QuantumTape): The tape to expand.
+        skip_first (Bool): If ``True``, will not expand a StatePrep operation if
+            it is the first operation in the tape.
+
+    Returns:
+        QuantumTape: The expanded version of ``tape``.
+
+    **Example**
+    ...
+    """
+    new_prep = []
+    new_ops = []
+
+    for i, op in enumerate(tape._prep + tape._ops):
+        if isinstance(op, StatePrep):
+            if i == 0 and skip_first:
+                new_prep.append(op)
+            else:
+                new_ops.extend(op.decomposition())
+        else:
+            new_ops.append(op)
+
+    # preserves inheritance structure
+    # if tape is a QuantumTape, returned object will be a quantum tape
```

Comment:
If the tape could be other things, then the Args section should be modified.

Reply:
Added 👍🏼

Reply:
Since QuantumTape is a QuantumScript, type hinting QuantumScript is enough.

Reply:
Swapped it to `QuantumScript`